lib/elixir/test/elixir/kernel/sigils_test.exs

```diff
@@ -121,6 +187,124 @@ bar) in ["foo\\\nbar", "foo\\\r\nbar"]
     assert ~W(Foo.Bar.Baz)a == [:"Foo.Bar.Baz"]
   end
 
+  test "sigil w/W warns on trailing comma at compile time, not runtime" do
+    assert capture_err(fn ->
+             Code.compile_string("""
+             defmodule Kernel.SigilsTest.Runtime_ws do
+               def run, do: ~w(foo, bar baz)s
+             end
+             """)
+           end) =~
+             "The sigils ~w/~W do not allow trailing commas"
+
+    assert capture_err(fn -> Kernel.SigilsTest.Runtime_ws.run() end) == ""
+
+    assert capture_err(fn ->
+             Code.compile_string("""
+             defmodule Kernel.SigilsTest.Runtime_wa do
+               def run, do: ~w(foo, bar baz)s
+             end
+             """)
+           end) =~
+             "The sigils ~w/~W do not allow trailing commas"
+
+    assert capture_err(fn -> Kernel.SigilsTest.Runtime_wa.run() end) == ""
+
+    assert capture_err(fn ->
+             Code.compile_string("""
+             defmodule Kernel.SigilsTest.Runtime_wc do
+               def run, do: ~w(foo, bar baz)s
+             end
+             """)
+           end) =~
+             "The sigils ~w/~W do not allow trailing commas"
+
+    assert capture_err(fn -> Kernel.SigilsTest.Runtime_wc.run() end) == ""
+
+    assert capture_err(fn ->
+             Code.compile_string("""
+             defmodule Kernel.SigilsTest.Runtime_Ws do
+               def run, do: ~w(foo, bar baz)s
+             end
+             """)
+           end) =~
+             "The sigils ~w/~W do not allow trailing commas"
+
+    assert capture_err(fn -> Kernel.SigilsTest.Runtime_Ws.run() end) == ""
+
+    assert capture_err(fn ->
+             Code.compile_string("""
+             defmodule Kernel.SigilsTest.Runtime_Wa do
+               def run, do: ~w(foo, bar baz)s
+             end
+             """)
+           end) =~
+             "The sigils ~w/~W do not allow trailing commas"
+
+    assert capture_err(fn -> Kernel.SigilsTest.Runtime_Wa.run() end) == ""
+
+    assert capture_err(fn ->
+             Code.compile_string("""
+             defmodule Kernel.SigilsTest.Runtime_Wc do
+               def run, do: ~w(foo, bar baz)s
+             end
+             """)
+           end) =~
+             "The sigils ~w/~W do not allow trailing commas"
+
+    assert capture_err(fn -> Kernel.SigilsTest.Runtime_Wc.run() end) == ""
+  end
+
+  test "sigil w/W warns on trailing comma inside macro" do
```

Comment:
I think there are no special conditions that would make this feature to behave differently if it's invoked inside a macro :)

I would remove this test, what do you think?

Reply:
There are no special conditions _today_. :)  I can remove it, though, your call.